util/check_tool_requirements.py

```diff
@@ -275,6 +275,23 @@ def to_semver(self, version, from_req):
         return '{}.{}{}'.format(major, minor, comb)
 
 
+class NinjaToolReq(ToolReq):
+    tool_cmd = ['ninja', '--version']
+
+    def to_semver(self, version, from_req):
+        # There exist different version string variants that we need to be
+        # able to parse. Some only contain the semantic version, e.g. "1.10.0",
+        # while others contain an additional suffix, e.g.
+        # "1.10.0.git.kitware.jobserver-1". This parser only extracts the first
+        # three digits and ignores the rest.
+        m = re.fullmatch(r'([0-9]+)\.([0-9]+)\.([0-9]+).*', version)
```

Comment:
Sorry for being late to the party. This looks good but, for next time, you might want just `re.match()`, since a `fullmatch` for `foo.*` is exactly the same thing as a `match` for `foo`. (I think?) Also, that would let you get rid of the capture groups because you could then just use `m.group(0)`.

Reply:
Yeah good point about the `.*` and `fullmatch`. I think initially wanted to use the capture groups for splitting the version numbers out - but since we're just joining them back together below, this is actually not needed.